tests/py/test_close.py

```diff
@@ -56,22 +66,85 @@ def test_can_post_to_close_page(self):
         assert response.headers['Location'] == '/~alice/'
         assert P('alice').is_closed
 
+
+    # balance checks
+
+    def test_close_page_is_not_available_to_owner_with_positive_balance(self):
+        self.make_participant('alice', claimed_time='now', last_paypal_result='', balance=10)
+        body = self.client.GET('/~alice/settings/close', auth_as='alice').body
+        assert 'Personal Information' not in body
+        assert 'You have a balance' in body
+
+    def test_close_page_is_not_available_to_owner_with_negative_balance(self):
+        self.make_participant('alice', claimed_time='now', last_paypal_result='', balance=-10)
+        body = self.client.GET('/~alice/settings/close', auth_as='alice').body
+        assert 'Personal Information' not in body
+        assert 'You have a balance' in body
+
+    def test_sends_owner_with_balance_and_no_paypal_on_quest_for_paypal(self):
+        self.make_participant('alice', claimed_time='now', balance=10)
+        body = self.client.GET('/~alice/settings/close', auth_as='alice').body
+        assert 'Personal Information' not in body
+        assert 'You have a balance' not in body
+        assert 'link a PayPal account' in body
+
+    def test_but_close_page_is_available_to_admin_even_with_positive_balance(self):
+        self.make_participant('admin', claimed_time='now', is_admin=True)
+        self.make_participant('alice', claimed_time='now', balance=10)
+        body = self.client.GET('/~alice/settings/close', auth_as='admin').body
+        assert 'Personal Information' in body
+        assert 'Careful!' in body
+
+    def test_and_close_page_is_available_to_admin_even_with_negative_balance(self):
+        self.make_participant('admin', claimed_time='now', is_admin=True)
+        self.make_participant('alice', claimed_time='now', balance=-10)
+        body = self.client.GET('/~alice/settings/close', auth_as='admin').body
+        assert 'Personal Information' in body
+        assert 'Careful!' in body
+
+    def test_posting_with_balance_fails_for_owner(self):
+        self.make_participant('alice', claimed_time='now', balance=10)
+        response = self.client.PxST('/~alice/settings/close', auth_as='alice')
+        assert response.code == 400
+        assert not P('alice').is_closed
+
+    def test_posting_with_balance_succeeds_for_admin(self):
```

Comment:
There's a principle in testing that making test case mimimal/simplest to achieve the test goal. Any additional conditions would add noise and unnecessery maintanance burden (consider other contributors having the same confusion as me). 

Of course that's in theory :)